testing/compile/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+thrust_update_system_found_flags()
+
+set(cmake_opts)
+set(extra_cuda_cmake_opts)
+set(extra_nvcxx_cmake_opts)
+
+set(cmake_opts
+  -D "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+  -D "CMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}"
+  -D "Thrust_DIR=${Thrust_SOURCE_DIR}/thrust/cmake"
+  -D "CMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}"
+)
+
+if (THRUST_CPP_FOUND AND THRUST_CUDA_FOUND)
+  # Test that we can use `find_package` on an installed Thrust:
+  set(extra_cuda_cmake_opts
+    -D "CMAKE_CUDA_COMPILER=${CMAKE_CUDA_COMPILER}"
+  )
+endif()
+
+# Need to pass these when testing NVC++.
+if ("NVCXX" STREQUAL "${CMAKE_CUDA_COMPILER_ID}")
+  set(extra_nvcxx_cmake_opts
+    -D "CMAKE_CUDA_COMPILER_ID=${CMAKE_CUDA_COMPILER_ID}"
+    -D "CMAKE_CUDA_COMPILER_FORCED=${CMAKE_CUDA_COMPILER_FORCED}"
+  )
+endif()
+
+# Add a build-and-test CTest.
+# - full_test_name_var will be set to the full name of the test.
+# - test_id is used to generate a unique name for this test.
+# - subdir is the relative path to the test project directory.
+# - Any additional args will be passed to the project configure step.
+function(thrust_add_compile_test full_test_name_var test_id subdir)
```

Comment:
It seems like the purpose of having `test_id` and `subdir` is to allow enumeration across the different execution policies.

It might be better to roll that into the function like so:

```cmake
function(thrust_add_compile_test full_test_name_var subdir exec_policy)
....

  thrust_add_compile_test(test_name
    fork_join_cdp_deprecation
    device
    "-D" "EXEC_POLICY=thrust::device"
    "-D" "IGNORE_DEPRECATION=FALSE"
   )
```